drivers/wdm/audio/backpln/portcls/pin_wavepci.cpp

```diff
@@ -588,22 +646,33 @@ CPortPinWavePci::DeviceIoControl(
 
     IoStack = IoGetCurrentIrpStackLocation(Irp);
 
-    if (IoStack->Parameters.DeviceIoControl.IoControlCode == IOCTL_KS_PROPERTY)
+    switch (IoStack->Parameters.DeviceIoControl.IoControlCode)
     {
-        return HandleKsProperty(Irp);
-    }
-    else if (IoStack->Parameters.DeviceIoControl.IoControlCode == IOCTL_KS_WRITE_STREAM || IoStack->Parameters.DeviceIoControl.IoControlCode == IOCTL_KS_READ_STREAM)
-    {
-       return HandleKsStream(Irp);
-    }
+        case IOCTL_KS_PROPERTY:
+            return HandleKsProperty(Irp);
 
-    UNIMPLEMENTED;
+        case IOCTL_KS_ENABLE_EVENT:
+            return PcHandleEnableEventWithTable(Irp, &m_Descriptor);
+
+        case IOCTL_KS_DISABLE_EVENT:
+            return PcHandleDisableEventWithTable(Irp, &m_Descriptor);
```

Comment:
Are you sure these new IOCTLs should be handled here? I see them handled already in `CPortPinWaveCyclic::DeviceIoControl`

Reply:
As I see, the code from pin_wavecyclic.cpp is not used at all in case with Vinyl AC97 driver.
So I'm not sure exactly about these two IOCTLs, but the driver definitely needs `IOCTL_KS_RESET` (it is called at least).
Although I think if the code from pin_wavepci.cpp and pin_wavecyclic.cpp is used separately for different audio drivers, then additional IOCTLs will not interfere in this case.